### ISSUE TITLE
Compiling 1.3 on Ubuntu 14.04

### ISSUE THREAD

**Issue by M4he** [h2o/h2o#376]
Hello, I'm trying to compile the H2O server on Ubuntu Server 14.04 via the following `cmake` command:
```
cmake -DWITH_BUNDLED_SSL=on -DCMAKE_INSTALL_PREFIX=/usr
```
It works fine for v1.2.0 but with v1.3.0 or v1.3.1 `make` stops with the following error:
```
In file included from .../include/h2o/multithread.h:27:0,
from .../include/h2o/hostinfo.h:32,
from .../lib/common/hostinfo.c:22:
.../include/h2o/socket.h:31:25: fatal error: openssl/ssl.h: No such file or directory
 #include <openssl/ssl.h>
```
(paths intentionally shortened)

**Comment by swar0g** [comment]
Did you installed  libssl-dev from repo or manually?

**Comment by M4he** [comment]
@swar0g I have not installed any libssl-dev package. The installation instructions state that at least OpenSSL 1.0.2 is needed because of the required ALPN support. Version 1.0.2 is not available for Ubuntu 14.04. Therefore I chose to use the bundled LibreSSL instead (`-DWITH_BUNDLED_SSL=on`).

I can compile and use H2O v1.2.0 without issues on this system using that flag.

**Comment by M4he** [comment]
I installed libssl-dev package and it seems to be working now. I still don't fully understand why OpenSSL is still needed as of 1.3 even when opting for the bundled LibreSSL.
